Add game controls example

colleenmcginnis opened the pull request:
Adds a new example for moving around the map with game-like controls as described in #2943  
Can you please review, @mollymerp and @lucaswoj ?

lucaswoj commented on docs/_posts/examples/3400-01-26-game-controls.html:
```diff
@@ -0,0 +1,133 @@
+    map.doubleClickZoom.disable();
+    map.scrollZoom.disable();
+    map.keyboard.disable();
+    map.touchZoomRotate.disable();
```
This block of code disables all interaction handlers except `dragRotate`. Do you intend to leave `dragRotate` enabled? Can you use the [`interactive: false` constructor option](https://www.mapbox.com/mapbox-gl-js/api/#Map) instead?

lucaswoj commented on docs/_posts/examples/3400-01-26-game-controls.html:
```diff
@@ -0,0 +1,133 @@
+        b[1].addEventListener('touchstart', buttonStart.bind(this, b));
+        b[1].addEventListener('mouseup', buttonEnd.bind(this, b));
+        b[1].addEventListener('touchend', buttonEnd.bind(this, b));
+    });
```
Do we really need buttons AND keyboard shortcuts? A lot of code could be eliminated & simplified if we dropped the buttons.

lucaswoj commented on docs/_posts/examples/3400-01-26-game-controls.html:
```diff
@@ -0,0 +1,133 @@
+                easing: easeTo
+            });
+        }
+    }
```
This `move` function does not provide any abstraction over the `Map#easeTo` method. Can we eliminate this function in favor of `Map#easeTo`? Note that `Map#panBy` calls `Map#easeTo` internally.

lucaswoj commented on docs/_posts/examples/3400-01-26-game-controls.html:
```diff
@@ -0,0 +1,133 @@
+---
+layout: example
+category: example
+title: Game-like controls
```
The example title should be a verb phrase

 > Navigate the map with game-like controls

colleenmcginnis commented:
Addressed @lucaswoj 's comments including taking away the buttons (so the user only navigates with the keyboard). Since you need to focus on the map before the arrow keys will work, I added one line of instruction on top of the map that disappears once the user clicks on the map. 
![image](https://cloud.githubusercontent.com/assets/10479155/18494762/3cdd5232-79cd-11e6-92ca-a7c244cbab6f.png)

mollymerp commented:
Nice work @colleenmcginnis! 

One note: you may want to get rid of `goDirection` entirely, and just put the `map.easeTo` and `map.panBy` calls inside the event listener function (where you currently call `goDirection()` )to make the example even more straight forward.

lucaswoj commented:
Just pushed up some minor comment edits 🙇 

This is ready to merge on 🍏!

mollymerp commented on docs/_posts/examples/3400-01-26-game-controls.html:
```diff
@@ -0,0 +1,81 @@
+            map.easeTo({
+                bearing: map.getBearing() - deltaDegrees,
+                easing: easeFunction,
+                animate: true
```
also you can remove `animate: true` on your `map.eastTo()` calls -- its `true` by defailt 😸

colleenmcginnis replied:
done @mollymerp !

mollymerp commented:
Awesome! Feel free to merge on 🍏 @colleenmcginnis 👏
